app/aws/PasskeyDB.scala

```diff
@@ -54,16 +77,136 @@ object PasskeyDB {
     val request =
       PutItemRequest.builder().tableName(tableName).item(item.asJava).build()
     dynamoDB.putItem(request)
-  }.map(_ => ())
-    .recoverWith(exception =>
+    ()
+  }.recoverWith(exception =>
+    Failure(
+      JanusException(
+        userMessage = "Failed to store passkey",
+        engineerMessage =
+          s"Failed to store credential for user ${user.username}: ${exception.getMessage}",
+        httpCode = INTERNAL_SERVER_ERROR,
+        causedBy = Some(exception)
+      )
+    )
+  )
+
+  def loadCredential(
+      user: UserIdentity,
+      credentialId: Array[Byte]
+  )(implicit dynamoDB: DynamoDbClient): Try[GetItemResponse] = {
+    Try {
+      val key = Map(
+        "username" -> AttributeValue.fromS(user.username),
+        "credentialId" -> AttributeValue.fromS(
+          Base64UrlUtil.encodeToString(credentialId)
+        )
+      )
+      val request =
+        GetItemRequest.builder().tableName(tableName).key(key.asJava).build()
+      dynamoDB.getItem(request)
+    }.recoverWith(err =>
       Failure(
         JanusException(
-          userMessage = "Failed to store credential",
+          userMessage = "Failed to find registered passkey",
           engineerMessage =
-            s"Failed to store credential for user ${user.username}: ${exception.getMessage}",
+            s"Failed to load credential for user ${user.username}: ${err.getMessage}",
           httpCode = INTERNAL_SERVER_ERROR,
-          causedBy = Some(exception)
+          causedBy = Some(err)
         )
       )
     )
+  }
+
+  def extractCredential(
+      response: GetItemResponse,
+      user: UserIdentity
+  ): Try[CredentialRecord] = {
+    if (response.hasItem) {
```

Comment:
Not super urgent, but I think I'd look at whether this if statement (and the resulting duplication of error handling at the bottom as well as the nesting it causes) could be inlined. Maybe making this a for comp and doing `item <- response.item().toRight(<err>).toTry` as the first line would make this simpler? Perhaps something to fiddle with when we do another pass on this.

Reply:
Thanks!  I'll keep a note of this and follow up